Tests/SonarScanner.MSBuild.PreProcessor.Test/JreCaching/JreCacheTests.cs

```diff
@@ -57,6 +62,17 @@ private static IEnumerable<object[]> DirectoryAndFileCreateAndMoveExceptions
         }
     }
 
+    public JreCacheTests()
+    {
+        testLogger = new TestLogger();
+        directoryWrapper = Substitute.For<IDirectoryWrapper>();
+        fileWrapper = Substitute.For<IFileWrapper>();
+        checksum = Substitute.For<IChecksum>();
+        unpack = Substitute.For<IUnpack>();
+        unpackProvider = Substitute.For<IUnpackProvider>();
+        unpackProvider.GetUnpackForArchive(directoryWrapper, fileWrapper, TestArchiveName).Returns(unpack);
+    }
```

Comment:
@gregory-paidis-sonarsource asked me to put in fields to avoid duplication here https://github.com/SonarSource/sonar-scanner-msbuild/pull/2016#discussion_r1665250040. In MSTest, you have one instance of the test class per test case invocation, so there is no state sharing. Putting the initialization in the ctor allows making the fields readonly. See also the discussion on SO: https://stackoverflow.com/a/45270180

Reply:
I did not know that there is one instance of the test class per test invocation.
If that is true, I think it is fine to leave it on the constructor.

Reply:
Ok. Then we're good for now. 

I personally prefer to have the fields set up independently. This way, we would not depend on the particularities of the test framework and could switch if a need arises (like we had to switch away from Moq)

Reply:
xUnit [behaves the same](https://xunit.net/docs/shared-context) and NUnit [is configurable](https://docs.nunit.org/articles/nunit/writing-tests/attributes/fixturelifecycle.html), so it should be portable.